store.go

```diff
@@ -3815,3 +3835,55 @@ func (s *store) GarbageCollect() error {
 
 	return firstErr
 }
+
+// List returns a ListResult structure that contains layer, image, or container
+// extracts according to the values in ListOptions.
+func (s *store) List(options ListOptions) (ListResult, error) {
```

Comment:
Throughout: If the relevant option is not set, it’s not necessary to lock the store.

That’s particularly important for the container store if we are not going to use it at all in the initial caller.